exec/java-exec/src/main/java/org/apache/drill/exec/store/ischema/RecordCollector.java

```diff
@@ -318,36 +319,39 @@ public MetastoreRecordCollector(Metastore metastore, FilterEvaluator filterEvalu
      * @param schemaPath schema name
      * @param table table instance
      * @param schema table or column schema
-     * @param parentColumnName parent column name if any
+     * @param parentColumnNames list of parent column names if any
      * @param columnIndex column index if any
      * @param isNested indicates if column is nested
      * @return list of column records
      */
     private List<Records.Column> columns(String schemaPath,
                                          BaseTableMetadata table,
                                          TupleMetadata schema,
-                                         String parentColumnName,
+                                         List<String> parentColumnNames,
```

Comment:
Should this be a list of names? Or a pointer to the parent? Seems redundant, for a name like `date[0].month` to have

```
([], 'date')
(`date`, `[]`)
(`date[]`, `month)
```

Also, would be helpful to spell out what the names include. Am I right in the above examples? Are the array brackets included?

Reply:
This is a list that contains unquoted names of columns parent to this column. For example, for column `a.b.c.d`, when a column `d` will be handled in this method, the list would contain (`a`, `b`, `c`) columns. Array brackets are ignored.